types/multer/index.d.ts

```diff
@@ -47,10 +48,22 @@ declare namespace multer {
     }
 
     interface StorageEngine {
-        _handleFile(req: express.Request, file: Express.Multer.File, callback: (error?: any, info?: Express.Multer.File) => void): void;
+        _handleFile(req: express.Request, file: Express.Multer.File, callback: (error?: any, info?: FileInfo) => void): void;
```

Comment:
It's better to use `Partial<Express.Multer.File>` here. See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html.
For this you have to append `// TypeScript Version: 2.1` line to header.